devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java

```diff
@@ -125,55 +122,41 @@ public QuarkusDev(
      * for up-to-date checks
      */
     @SuppressWarnings("unused")
-    @CompileClasspath
+    @Internal
     public Configuration getQuarkusDevConfiguration() {
         return this.quarkusDevConfiguration;
     }
 
-    /**
-     * The JVM sources (Java, Kotlin, ..) for the project
-     */
-    @Optional
-    @InputFiles
-    @PathSensitive(PathSensitivity.RELATIVE)
-    public FileCollection getSources() {
-        return mainSourceSet.getAllJava().getSourceDirectories();
-    }
-
-    /**
-     * The JVM classes directory (compilation output)
-     */
-    @Optional
-    @InputFiles
-    @PathSensitive(PathSensitivity.RELATIVE)
-    public FileCollection getCompilationOutput() {
-        return mainSourceSet.getOutput().getClassesDirs();
-    }
-
     /**
      * The directory to be used as the working dir for the dev process.
      *
      * Defaults to the main source set's classes directory. If there are
      * multiple, one is picked at random (see {@link QuarkusPluginExtension#getLastFile}).
      */
-    @Input
+    @Internal
     public Property<File> getWorkingDirectory() {
         return workingDirectory;
     }
 
     /**
-     * @deprecated See {@link #workingDirectory}
+     * @see #workingDirectory
      */
-    @Deprecated
     public void setWorkingDir(String workingDir) {
```

Comment:
shouldn't we keep it as deprecated?

Reply:
Up to y'all.  I added the deprecation in the first place.  Depends how you want upgrading users to experience this.  

Again, I'll do whatever you want here :)